tsl/src/bgw_policy/job_api.c

```diff
@@ -201,10 +201,20 @@ static BgwJob *
 find_job(int32 job_id, bool null_job_id, bool missing_ok)
 {
 	BgwJob *job;
+	bool got_lock;
+	LOCKTAG tag;
 
 	if (null_job_id && !missing_ok)
 		ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE), errmsg("job ID cannot be NULL")));
 
+	/* take advisory lock before relation lock */
```

Comment:
Could you extend this comment a bit? Currently, the motivation for the lock is only stated in the commit message. 
Without knowing the commit message, the motivation for the second lock might be hard to understand. Maybe you could take a part of the commit message and paste it here.

Reply:
Good point. Will do.

Reply:
Done